### PR TITLE
Monkey mind mag helmets's sprites don't break on updating their icon_state

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

adds a special override for update icon that keeps the light color.

## Why It's Good For The Game

ERROR ERROR

## Changelog
:cl:
fix: mind mag helms losing their icon sometimes
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
image

**Comment by yeeyeh** [comment]
munkee

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -346,10 +346,12 @@
 	strip_delay = 100
 	var/mob/living/carbon/monkey/magnification = null ///if the helmet is on a valid target (just works like a normal helmet if not (cargo please stop))
 	var/polling = FALSE///if the helmet is currently polling for targets (special code for removal)
+	var/light_colors = 1 ///which icon state color this is (red, blue, yellow)
 
 /obj/item/clothing/head/helmet/monkey_sentience/Initialize()
 	. = ..()
-	icon_state = "[icon_state][rand(1,3)]"
+	light_colors = rand(1,3)
+	icon_state = "[icon_state][light_colors]"
```
Comment (ShizCalev) [inline code/modules/clothing/head/helmet.dm:354]: call update_icon instead

Comment (tralezab) [reply code/modules/clothing/head/helmet.dm:354]: smart!

**Comment by ShizCalev** [comment]
misclick, what did you say

File: code/modules/clothing/head/helmet.dm (hunk 1 above)
Comment (ShizCalev) [inline code/modules/clothing/head/helmet.dm:354]: call update_icon instead

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -360,6 +362,10 @@
 	. += "<span class='warning'>GENETIC MAKEUP MASS SUSCEPTIBILITY</span>"
 	. += "<span class='boldnotice'>Ask your CMO if mind magnification is right for you.</span>"
 
+/obj/item/clothing/head/helmet/monkey_sentience/update_icon_state()
+	var/lightson = magnification ? "up" : ""
```
Comment (ShizCalev) [inline code/modules/clothing/head/helmet.dm:366]: just do this in the line below

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by tralezab** [comment]
👍

**Pull request merged**